src/Fixer/Casing/MagicConstantCasingFixer.php

```diff
@@ -77,6 +78,10 @@ private function getMagicConstants()
             if (defined('T_TRAIT_C')) {
                 $magicConstants[T_TRAIT_C] = '__TRAIT__';
             }
+
+            if (PHP_VERSION_ID >= 50500) {
```

Comment:
it cheaper to the test here one time and when not hit not add it to the array, that way you won't compare all the tokens against a type you already know is never available.